src/attr/_make.py

```diff
@@ -2724,6 +2723,21 @@ def __init__(self, converter, *, takes_self=False, takes_field=False):
             converter
         ).get_first_param_type()
 
+    def __call__(self, value: object, instance: object, field: Attribute):
+        """
+        Call the converter with the appropriate subset of *instance* and
+        *field*.
+        """
+        return self.converter(
+            value,
+            *{
```

Comment:
This dict thing is cute but it's going to be slower than some if statements, and less readable :/

Reply:
Can we do this decision at Converter instantiation time instead of calling time?